CHANGELOG.md

```diff
@@ -125,9 +125,13 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 - Added `Accelerator.is_available` to check device availability ([#11797](https://github.com/PyTorchLightning/pytorch-lightning/pull/11797))
 
 
+- Enabled static type-checking on the signature of `Trainer` ([#11888](https://github.com/PyTorchLightning/pytorch-lightning/pull/11888))
```

Comment:
do we need a changelog entry for this?

Reply:
I wouldn't have asked for it but it doesn't hurt :)